File: typescript/src/api/resources/changesets/client/requests/CreateChangeSetRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         publicSummary: "publicSummary",
 *         publicTitle: "publicTitle",
 *         scopeRef: "scopeRef",
 *         scopeType: "global",
 *         sourceType: "import"
 *     }
 */
export interface CreateChangeSetRequest {
    agentId?: string | null;
    baseReleaseId?: string | null;
    candidateReleaseId?: string | null;
    publicSummary: string;
    publicTitle: string;
    scopeRef: string;
    scopeType: Sikaru.ScopeType;
    sourceType: Sikaru.SourceType;
    workflowId?: string | null;
}

File: ruby/lib/Sikaru/changesets/types/create_change_set_request.rb
# frozen_string_literal: true

module Sikaru
  module Changesets
    module Types
      class CreateChangeSetRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :agent_id, -> { String }, optional: true, nullable: false, api_name: "agentId"

        field :base_release_id, -> { String }, optional: true, nullable: false, api_name: "baseReleaseId"

        field :candidate_release_id, -> { String }, optional: true, nullable: false, api_name: "candidateReleaseId"

        field :public_summary, -> { String }, optional: false, nullable: false, api_name: "publicSummary"

        field :public_title, -> { String }, optional: false, nullable: false, api_name: "publicTitle"

        field :scope_ref, -> { String }, optional: false, nullable: false, api_name: "scopeRef"

        field :scope_type, -> { Sikaru::Types::ScopeType }, optional: false, nullable: false, api_name: "scopeType"

        field :source_type, -> { Sikaru::Types::SourceType }, optional: false, nullable: false, api_name: "sourceType"

        field :workflow_id, -> { String }, optional: true, nullable: false, api_name: "workflowId"
      end
    end
  end
end
